backend/utils/auth.py

```
@@ -0,0 +1,34 @@
+    Validates the Bearer token and returns the user object.
+    """
+    try:
+        # Get the user using the token
+        user_response = supabase.auth.get_user(token.credentials)
+        
+        if not user_response.user:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Invalid authentication credentials",
+                headers={"WWW-Authenticate": "Bearer"},
+            )
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)